File: RealTimeFaceDetection.py
import cv2
import numpy as np

cap = cv2.VideoCapture(0)
facecascade = cv2.CascadeClassifier("Resources/haarcascade_frontalface_default.xml")
# cap.set(3,640)
# cap.set(4,480)
cap.set(10,100) #For brightness
count = 0
while True:
    success, img = cap.read()
    imgGray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
    faces = facecascade.detectMultiScale(imgGray, 1.1, 10)
    for (x, y, w, h) in faces:
        area = w*h
        if area>500:
            cv2.rectangle(img, (x, y), (x + w, y + h), (255, 0, 255), 2)
            cv2.putText(img,"Face Detected",(x,y-5),cv2.FONT_HERSHEY_SIMPLEX,1,(255,0,255),1)
            imgRoi = img[y:y+h,x:x+w]
            cv2.imshow("Faces", imgRoi)

    cv2.imshow("Result", img)
    if cv2.waitKey(1) == ord('s'):
        cv2.imwrite("Resources/Scanned/Face_"+str(count)+".jpg",imgRoi)
        cv2.rectangle(img,(0,200),(640,300),(0,255,0),cv2.FILLED)
        cv2.putText(img,"Scanned",(150,265),cv2.FONT_HERSHEY_DUPLEX,
                    2,(0,0,255),2)
        cv2.imshow("Result",img)
        cv2.waitKey(500)
        count+=1

        break

File: RealTimeFaceDetection.cpp
#include <opencv2/imgcodecs.hpp>
#include <opencv2/highgui.hpp>
#include <opencv2/imgproc.hpp>
#include <opencv2/objdetect.hpp> //Used for detecting face/eyes.....//
#include <iostream>

using namespace cv;
using namespace std;


///////////////  Face Detection  //////////////////////

int main() {

    VideoCapture cap(0);
    Mat img;


    CascadeClassifier faceCascade;
    faceCascade.load("Resources/haarcascade_frontalface_default.xml");
    if (faceCascade.empty()) { cout << "XML file is not loaded" << endl; }
    vector<Rect> faces;

    while (true) {

        cap.read(img);
        faceCascade.detectMultiScale(img, faces, 1.1, 10);

        for (int i = 0; i < faces.size(); i++) {

            //Cropping and saving the face//

            Mat imgCrop = img(faces[i]);
            imwrite("Resources/faces/" + to_string(i) + ".png", imgCrop);
            

            rectangle(img, faces[i].tl(), faces[i].br(), Scalar(255, 0, 255), 3);
            imshow("Image Crop", imgCrop);


        }



        imshow("Image", img);
      
        waitKey(1); //This adds a delay where '0' means infinity otherwise the image will close automatically//

    }
    return 0;
}

